vault/token_store.go

```diff
@@ -1010,6 +1023,84 @@ func (ts *TokenStore) lookupBySaltedAccessor(saltedAccessor string) (accessorEnt
 	return aEntry, nil
 }
 
+// handleCleanup handles the cleaning up of leaked accessor storage entries and
+// cleaning up of leases that are associated to tokens that are expired.
+func (ts *TokenStore) handleCleanup(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	// List out all the accessors
+	saltedAccessorList, err := ts.view.List(accessorPrefix)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch accessor entries: %v", err)
+	}
+
+	var cleanupErrors *multierror.Error
+
+	log.Printf("len(saltedAccessorList): %d\n", len(saltedAccessorList))
+
+	// For each of the accessor, see if the token ID associated with it is
+	// a valid one. If not, delete the leases associated with that token
+	// and delete the accessor as well.
+	for _, saltedAccessor := range saltedAccessorList {
+		accessorEntry, err := ts.lookupBySaltedAccessor(saltedAccessor)
+		if err != nil {
+			cleanupErrors = multierror.Append(cleanupErrors, fmt.Errorf("failed to read the accessor entry: %v", err))
+			continue
+		}
+
+		if accessorEntry.TokenID == "" {
```

Comment:
Please add a comment above to make it clear why the rest is being skipped -- that this entry itself is bogus.